File: veterinarskaStanicaBackend/eVeterinarskaStanicaModel/Pet.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace eVeterinarskaStanicaModel
{
    public enum PetGender
    {
        Male = 1,
        Female = 2,
    }

    public enum PetStatus
    {
        Active = 1,
        Inactive = 2,
        Deceased = 3
    }

    public class Pet
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string Species { get; set; } = string.Empty; // Dog, Cat, Bird, etc.

        [StringLength(50)]
        public string? Breed { get; set; }

        public PetGender Gender { get; set; }

        public DateTime? DateOfBirth { get; set; }

        [StringLength(50)]
        public string? Color { get; set; }

        public decimal? Weight { get; set; } // in kg

        [StringLength(50)]
        public string? MicrochipNumber { get; set; }

        public PetStatus Status { get; set; } = PetStatus.Active;

        [StringLength(1000)]
        public string? Notes { get; set; }

        [StringLength(500)]
        public string? PhotoUrl { get; set; }

        public DateTime DateCreated { get; set; } = DateTime.UtcNow;

        public DateTime? DateModified { get; set; }

        // Foreign Keys
        public int PetOwnerId { get; set; }
        
        // Ko je dodao pacijenta (veterinar ili admin)
        public int? CreatedBy { get; set; }

        // Navigation Properties
        public virtual User PetOwner { get; set; }
        public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
        public virtual ICollection<MedicalRecord> MedicalRecords { get; set; } = new List<MedicalRecord>();
        public virtual ICollection<Order> Orders { get; set; } = new List<Order>();
        public virtual ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
        public virtual ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();
        public virtual ICollection<Notification> Notifications { get; set; } = new List<Notification>();
    }
}

File: lib/models/pet.dart
import 'package:json_annotation/json_annotation.dart';
import 'user.dart';

part 'pet.g.dart';

enum PetGender {
  @JsonValue(1)
  male,
  @JsonValue(2)
  female,
}

enum PetStatus {
  @JsonValue(1)
  active,
  @JsonValue(2)
  inactive,
  @JsonValue(3)
  deceased,
}

@JsonSerializable()
class Pet {
  final int id;
  final String name;
  final String species;
  final String? breed;
  final PetGender gender;
  final DateTime? dateOfBirth;
  final String? color;
  final double? weight;
  final String? microchipNumber;
  final PetStatus status;
  final String? notes;
  final String? photoUrl;
  final DateTime dateCreated;
  final DateTime? dateModified;
  final int petOwnerId;
  final User? petOwner;

  Pet({
    required this.id,
    required this.name,
    required this.species,
    this.breed,
    required this.gender,
    this.dateOfBirth,
    this.color,
    this.weight,
    this.microchipNumber,
    required this.status,
    this.notes,
    this.photoUrl,
    required this.dateCreated,
    this.dateModified,
    required this.petOwnerId,
    this.petOwner,
  });

  factory Pet.fromJson(Map<String, dynamic> json) => _$PetFromJson(json);

  Map<String, dynamic> toJson() => _$PetToJson(this);

  // Helper getters for UI
  String get genderText {
    switch (gender) {
      case PetGender.male:
        return 'Muški';
      case PetGender.female:
        return 'Ženski';
    }
  }

  String get statusText {
    switch (status) {
      case PetStatus.active:
        return 'Aktivan';
      case PetStatus.inactive:
        return 'Neaktivan';
      case PetStatus.deceased:
        return 'Preminuo';
    }
  }

  String get ageText {
    if (dateOfBirth == null) return 'Nepoznato';
    
    final now = DateTime.now();
    final age = now.difference(dateOfBirth!);
    final years = age.inDays ~/ 365;
    final months = (age.inDays % 365) ~/ 30;
    
    if (years > 0) {
      return '$years godina${months > 0 ? ' i $months meseci' : ''}';
    } else if (months > 0) {
      return '$months meseci';
    } else {
      return '${age.inDays} dana';
    }
  }

  String get ownerName => petOwner?.firstName != null && petOwner?.lastName != null
      ? '${petOwner!.firstName} ${petOwner!.lastName}'
      : petOwner?.email ?? 'Nepoznato';
}










